tests/test_extensions/test_ext_intersphinx.py

```diff
@@ -745,30 +745,51 @@ def test_intersphinx_role(app):
 
 
 @pytest.mark.sphinx('html', testroot='root')
-def test_intersphinx_cache_limit(app):
+@pytest.mark.parametrize(
+    ('cache_limit', 'expected_expired'), [(5, False), (1, True), (-1, False)]
+)
+def test_intersphinx_cache_limit(app, monkeypatch, cache_limit, expected_expired):
     url = 'https://example.org/'
-    app.config.intersphinx_cache_limit = -1
+    app.config.intersphinx_cache_limit = cache_limit
     app.config.intersphinx_mapping = {
         'inv': (url, None),
     }
     # load the inventory and check if it's done correctly
     intersphinx_cache: dict[str, InventoryCacheEntry] = {
-        url: ('', 0, {}),  # 0 is a timestamp, make sure the entry is expired
+        url: ('inv', 0, {}),  # Timestamp of last cache write is zero.
     }
     validate_intersphinx_mapping(app, app.config)
-    load_mappings(app)
 
-    now = int(time.time())
+    # The test's `now` is two days after the cache was created.
+    now = 2 * 86400
+    monkeypatch.setattr('time.time', mock.Mock(name='time.time', return_value=now))
+
+    # `_fetch_inventory_group` calls `_fetch_inventory`. We replace it
+    # with a mock to test whether it has been called or not. If it has
+    # been called, it means the cache had expired.
+    mock_fetch_inventory = mock.Mock(return_value=('inv', now, {}))
+    monkeypatch.setattr(
+        'sphinx.ext.intersphinx._load._fetch_inventory', mock_fetch_inventory
+    )
+
     for name, (uri, locations) in app.config.intersphinx_mapping.values():
         project = _IntersphinxProject(name=name, target_uri=uri, locations=locations)
-        # no need to read from remote
-        assert not _fetch_inventory_group(
+        updated = _fetch_inventory_group(
             project=project,
             cache=intersphinx_cache,
             now=now,
             config=app.config,
             srcdir=app.srcdir,
         )
+        # If we hadn't mocked `_fetch_inventory`, it would've made
```

Comment:
One way to make it work is to have a fake HTTP server serving the inventory as we usually do. It's my bad that I didn't think of it when I said that the test was fine.

Reply:
@AA-Turner Do you have a preference for this test or not? we can either create a fake HTTP server for the test or mock the tests like this. With an HTTP server, I feel that we wouldn't rely on possible changes of `_fetch_inventory` for whatever reason.